project/src/main/java/com/google/sps/servlets/CreateBusinessAccountServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.google.sps.servlets;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+@WebServlet("/createBusinessAccount")
+public class CreateBusinessAccountServlet extends HttpServlet {
+
+  protected DatastoreService datastore;
+  protected UserService userService;
+
+  public CreateBusinessAccountServlet() {
+    super();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    userService = UserServiceFactory.getUserService();
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get required user information.
+    String userId = userService.getCurrentUser().getUserId();
+    String userEmail = userService.getCurrentUser().getEmail();
+    String nickname = request.getParameter("nickname"); 
+    String street = request.getParameter("street");
+    String city = request.getParameter("city");
+    String state = request.getParameter("state");
+    String zipCode = request.getParameter("zipCode");
+    List<String> searchHistory = new ArrayList<>(); // no search history initally
```

Comment:
Will you implement the search history for business accounts in addition to consumers? (It's fine either way, but I only remember seeing notes about implementing it for the consumer accounts).

Reply:
I'm treating business accounts as normal users with extra permissions, so the person who owns the business account can browse the website normally as well.